integrations/venice-duckdb/src/main/java/com/linkedin/venice/sql/DuckDBDaVinciRecordTransformer.java

```diff
@@ -0,0 +1,130 @@
+        .name("lastName")
+        .type()
+        .stringType()
+        .stringDefault("")
+        .endRecord();
+    return valueSchema;
+  }
+
+  @Override
+  public DaVinciRecordTransformerResult<GenericRecord> transform(Lazy<String> key, Lazy<GenericRecord> value) {
+    // Record transformation happens inside processPut as we need access to the connection object to create the prepared
+    // statement
+    return new DaVinciRecordTransformerResult<>(DaVinciRecordTransformerResult.Result.UNCHANGED);
+  }
+
+  @Override
+  public void processPut(Lazy<String> key, Lazy<GenericRecord> value) {
+    Schema valueSchema = value.get().getSchema();
+    String upsertStatement = AvroToSQL.upsertStatement(versionTableName, valueSchema);
+
+    try (Connection connection = DriverManager.getConnection(duckDBUrl)) {
+      BiConsumer<GenericRecord, PreparedStatement> upsertProcessor = AvroToSQL.upsertProcessor(valueSchema);
+
+      try (PreparedStatement preparedStatement = connection.prepareStatement(upsertStatement)) {
+        upsertProcessor.accept(value.get(), preparedStatement);
+      }
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void processDelete(Lazy<String> key) {
+    try (Connection connection = DriverManager.getConnection(duckDBUrl);
+        Statement stmt = connection.createStatement()) {
+      String sqlDelete = String.format("DELETE FROM %s WHERE %s = '%s';", versionTableName, primaryKey, key.get());
+      stmt.execute(sqlDelete);
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void onStartVersionIngestion(boolean isCurrentVersion) {
+    try (Connection connection = DriverManager.getConnection(duckDBUrl);
+        Statement stmt = connection.createStatement()) {
+      String createTableStatement =
+          AvroToSQL.createTableStatement(versionTableName, getOutputValueSchema(), primaryKeys, FAIL);
+      stmt.execute(createTableStatement);
+
+      if (isCurrentVersion) {
+        String createViewStatement =
+            String.format("CREATE OR REPLACE VIEW current_version AS SELECT * FROM %s;", versionTableName);
+        stmt.execute(createViewStatement);
+      }
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void onEndVersionIngestion(int currentVersion) {
+    try (Connection connection = DriverManager.getConnection(duckDBUrl);
+        Statement stmt = connection.createStatement()) {
+      // Swap to current version
+      String currentVersionTableName = baseVersionTableName + currentVersion;
+      String createViewStatement =
+          String.format("CREATE OR REPLACE VIEW current_version AS SELECT * FROM %s;", currentVersionTableName);
+      stmt.execute(createViewStatement);
+
+      // Drop DuckDB table for storeVersion as it's retired
+      String deleteTableStatement = String.format("DROP TABLE %s;", versionTableName);
```

Comment:
ditto for this one.

https://xkcd.com/327/

Reply:
We should have a unit test called `littleBobbyTables()`